models/dataset.go

```diff
@@ -457,17 +457,49 @@ func ValidateDataset(ctx context.Context, dataset *Dataset) error {
 	var invalidFields []string
 	if dataset.URI != "" {
 		dataset.URI = strings.TrimSpace(dataset.URI)
-		datasetURI, err := url.Parse(dataset.URI)
+		_, err := url.Parse(dataset.URI)
 		if err != nil {
 			invalidFields = append(invalidFields, "URI")
 			log.Event(ctx, "error parsing URI", log.ERROR, log.Error(err))
-		} else {
-			if !strings.Contains(datasetURI.Path, "/datasets") || !strings.Contains(datasetURI.Path, dataset.ID) {
-				fmt.Println("hello")
-				invalidFields = append(invalidFields, "URI")
-			}
 		}
 	}
+
+	for i := range dataset.Publications {
+		generalDetails := &dataset.Publications[i]
+		if generalDetails != nil {
+			trimHref(ctx, generalDetails)
+		}
+	}
+
+	for i := range dataset.Methodologies {
+		generalDetails := &dataset.Methodologies[i]
+		if generalDetails != nil {
+			trimHref(ctx, generalDetails)
+		}
+	}
+
+	for i := range dataset.RelatedDatasets {
+		generalDetails := &dataset.RelatedDatasets[i]
+		if generalDetails != nil {
+			trimHref(ctx, generalDetails)
+		}
+	}
+
+	if invalidFields != nil {
+		return fmt.Errorf("invalid fields: %v", invalidFields)
+	}
+	return nil
+
+}
+
+func trimHref(ctx context.Context, generalDetails *GeneralDetails) error {
```

Comment:
Consider updating this to accept a `[]*GeneralDetails` you can do the iteration over the slice inside this function. Saves duplicating the `for`  logic.